### PR TITLE
 Migrating from Xerces-C++ 2.8.0 to Xerces-C++ 3.x

### PR DESCRIPTION
**Pull request by ianna**
Xerces-C++ 3.0.0 is a major release and includes a number of application-breaking interface changes compared to Xerces-C++ 2 series. The following sub-sections provide an overview of the public API changes between Xerces-C++ 2 series and this release.
- A number of DOM interfaces (DOMLSInput, DOMLSOuput, DOMLSParser, DOMLSSerializer, DOMConfiguration, etc.) were added as part of the the final DOM Level 3 specification conformance work
- Many public interfaces that used int/long types to represent memory-related sizes, counts, indexes, etc., have been modified to use the 64-bit safe XMLSize_t type instead
-  All APIs marked as deprecated in Xerces-C++ 2 series have been removed in this release. In particular deprecated DOM (depdom) as well as COM support have been removed.

Furthermore, a number of DOM interfaces (DOMBuilder, DOMWriter, DOMInputSource, etc.) were replaced as part of the the final DOM Level 3 specification conformance work.

### CODE REVIEW

**Comment by ianna** [comment]
@iahmad-khan and @smuzaffar - FYI

**Comment by ianna** [comment]
@smuzaffar - please, let me know if we decide to preserve backwards compatibility, I'd need to do the following modifications to the code:

```
#if _XERCES_VERSION <30100
#include "xercesc/dom/DOMWriter.hpp"
#else
 #include "xercesc/dom/DOMLSSerializer.hpp"
 #include "xercesc/dom/DOMLSOutput.hpp"
#endif
```

**Comment by smuzaffar** [comment]
cmsbuild, please test with cms-sw/cmsdist#2305

File: GeneratorInterface/LHEInterface/src/LHEReader.cc [hunk 1]
```diff
@@ -127,8 +127,9 @@ class LHEReader::XMLHandler : public XMLDocument::Handler {
 	                const XMLCh *const localname,
 	                const XMLCh *const qname) override;
 
-	void characters(const XMLCh *const data, const unsigned int length) override;
-	void comment(const XMLCh *const data, const unsigned int length) override;
+        virtual void characters (const XMLCh *const chars, const XMLSize_t length) override;
+        virtual void ignorableWhitespace (const XMLCh *const chars, const XMLSize_t length) override;
```
Comment (smuzaffar) [inline GeneratorInterface/LHEInterface/src/LHEReader.cc]: @Ianna, looks like ignorableWhitespace definition is missing?

**Comment by ianna** [comment]
please test

**Comment by smuzaffar** [comment]
cmsbuild, please test with cms-sw/cmsdist#2305

it needs to be tested with cmsdist :-)

**Pull request closed without merging**